pkg/hub/submarinerbroker/controller.go

```diff
@@ -238,6 +242,41 @@ func (c *submarinerBrokerController) createIPSecPSKSecret(ctx context.Context, b
 	return err
 }
 
+func isClusterSetEmpty(clusterSet *clusterv1beta2.ManagedClusterSet) bool {
+	for _, v := range clusterSet.Status.Conditions {
+		if v.Type == clusterv1beta2.ManagedClusterSetConditionEmpty {
+			return v.Status == metav1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+func (c *submarinerBrokerController) deleteBrokerResources(ctx context.Context, clusterSet *clusterv1beta2.ManagedClusterSet,
+	recorder events.Recorder,
+) error {
+	brokerNS := clusterSet.GetAnnotations()[SubmBrokerNamespaceKey]
+	if brokerNS == "" {
+		return nil
+	}
+	// TODO: Ideally, submariner finalizer check in caller should be enough and
+	//       we should just delete resources unconditionally if here. But we need
+	//       this check coz we're adding finalizer to non-submariner clustersets too
```

Comment:
I'm not sure I follow this. What does the finalizer have to do with deleting the broker NS? `resource.DeleteFromManifests` handles deleting non-existent resource.

Reply:
Idea is to use presence of finalizer as a check for submariner being installed on this clusterset and requiring cleanup or not. If we were to add finalizer only when installing submarineraddon, this check would be sufficent without needing to get manifestworks, namespace or anything else.

Reply:
I still don't see why you need to explicitly check if the `Namespace` exists before calling `resource.DeleteFromManifests` since `resource.DeleteFromManifests` handles `IsNotFound` error anyway.